src/layers/domain/response/response_models.py

```diff
@@ -0,0 +1,13 @@
+from typing import Generic, List, TypeVar
+
+from domain.core.aggregate_root import AggregateRoot
+
+T = TypeVar("T", bound=AggregateRoot)
+
+
+class SearchResponse(Generic[T], AggregateRoot):
+    results: List[T]
+
+    @classmethod
+    def from_models(cls, models: List[T]) -> "SearchResponse[T]":
+        return cls(results=models)
```

Comment:
from python 3.12 the typing syntax has dropped a lot of the need for imports

```suggestion
from domain.core.aggregate_root import AggregateRoot

class SearchResponse[T](AggregateRoot):
    results: list[T]

    @classmethod
    def from_models(cls, models: list[T]) -> "SearchResponse[T]":
        return cls(results=models)
```